web_external/js/views/body/UploadDatasetView.js

```diff
@@ -0,0 +1,276 @@
+            } else {
+                this.$('#isic-dataset-attribution-name').prop('disabled', false);
+            }
+        },
+        'change #isic-dataset-license': function (event) {
+            // Disable anonymous attribution unless CC-0 license is selected
+            var target = $(event.target);
+            var anonymous = this.$('#isic-dataset-attribution-anonymous');
+            if (target.val() === 'CC-0') {
+                anonymous.prop('disabled', false);
+            } else {
+                if (anonymous.prop('checked')) {
+                    this.$('#isic-dataset-attribution-attributed-to').prop('checked', true);
+                    this.$('#isic-dataset-attribution-attributed-to').change();
+                }
+                anonymous.prop('disabled', true);
+            }
+        },
+        'submit #isic-dataset-form': function (event) {
+            event.preventDefault();
+            this.$('#isic-dataset-submit').prop('disabled', true);
+            this.submitDataset();
+        }
+    },
+
+    initialize: function (settings) {
+        girder.cancelRestRequests('fetch');
+
+        this.uploadedZipFiles = [];
+        this.uploadedCsvFiles = [];
+        this.uploadFolderId = null;
+
+        this.render();
+    },
+
+    render: function () {
+        this.$el.html(isic.templates.uploadDataset({
+            user: girder.currentUser
+        }));
+
+        if (!this.uploadWidget) {
+            this.initializeUploadWidget();
+        }
+        this.updateUploadWidget();
+
+        this.$('input#isic-dataset-name').focus();
+
+        return this;
+    },
+
+    initializeUploadWidget: function () {
+        if (this.uploadWidget) {
+            this.stopListening(this.uploadWidget);
+            this.uploadWidget.destroy();
+        }
+        this.uploadWidget = new girder.views.UploadWidget({
+            parentView: this,
+            modal: false,
+            noParent: true,
+            title: false,
+            overrideStart: true
+        });
+
+        this.uploadWidget.setElement(this.$('.isic-upload-widget-container'));
+
+        this.listenTo(this.uploadWidget, 'g:filesChanged', this.filesSelected);
+        this.listenTo(this.uploadWidget, 'g:uploadStarted', this.uploadStarted);
+        this.listenTo(this.uploadWidget, 'g:uploadFinished', this.uploadFinished);
+    },
+
+    filesSelected: function (files) {
+        // TODO: could validate based on file extension
+    },
+    uploadStarted: function (files) {
+        // Prepare isic_upload folder in user's home and start upload
+        var uploadFolder = null;
+        var folderCollection = new girder.collections.FolderCollection();
+        folderCollection.once('g:changed', function () {
+            if (folderCollection.isEmpty()) {
+                // Create isic_upload folder
+                uploadFolder = new girder.models.FolderModel({
+                    name: 'isic_upload',
```

Comment:
We should append the current datetime to this folder name, to allow multiple upload files to remain for debugging, but not collide.